site/app/libraries/FileUtils.php

```diff
@@ -670,4 +670,30 @@ public static function checkForPermissionErrors(string $path, ?string $expected_
 
         return $results;
     }
+
+    /**
+     * Recursively traverse a directory structure starting at $dir.  The passed in $results array will be populated
+     * with the absolute path to each file or sub-directory.
+     *
+     * NOTE: Sub-directories are treated like files, and thus will be included as their own entries in the array.
+     *
+     * Credit:
+     * https://stackoverflow.com/questions/24783862/list-all-the-files-and-folders-in-a-directory-with-php-recursive-function
+     *
+     * @param string $dir The starting directory (not included in results)
+     * @param array $results An array passed by reference which will be populated
+     */
+    public static function getDirContents(string $dir, &$results = array()): void {
```

Comment:
What's the difference between this function and `getAllFiles` which is declared at the top of this file?
They look like they do the same thing